deploy/templates/gcs-manifests/gcs-mixins.yml.j2

```diff
@@ -0,0 +1,79 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: gluster-mixins
+  namespace: {{ monitoring_namespace }}
+spec:
+  template:
+    spec:
+      containers:
+      - name: gluster-mixins
+        image: quay.io/gluster/gluster-mixins:latest
+      restartPolicy: Never
+      serviceAccountName: gluster-mixins
+---
+kind: ServiceAccount
+apiVersion: v1
+metadata:
+  name: gluster-mixins
+  namespace: {{ monitoring_namespace }}
+---
+kind: ClusterRole
```

Comment:
Are all these permissions really required? Items that look suspicious:
- It's a ClusterRole, not just a Role
- Ability to '*' customresourecdefinitions
- Permissions on finalizers of prometheus & alertmanager
- Permission to create & list pods
- Permissions on namespaces

Reply:
Looking at the current container, the `manifests` directory has 3 files that create/modify:
- A namespace: "monitoring"
- A ConfigMap: "monitoring/grafana-dashboard-k8s-storage-resources-glusterfs-pv"
- A monitoring.coreos.com/v1 PrometheusRule: "monitoring/prometheus-alert-rules"

We're really just adding the ConfigMap and PrometheusRule to the current namespace (this Job is running in the monitoring namespace), so the `00namespace-namespace.yaml` (and associated namespace manipulation perms) shouldn't be needed. The `namespace` fields can also be removed from the other object definitions... This will make it possible to change the namespace the objects get deployed in (determined by the namespace where the Job is run).

It should then be possible to convert to a Role w/ just '*' on ConfigMap and PrometheusRule.

(the above will need to be verified, of course... I have likely missed something)

Reply:
In thinking more about this, for the above to work, we'll need to tell `kubectl` the correct namespace via an env variable:
```yaml
  template:
    spec:
      containers:
          env:
            - name: NAMESPACE
              valueFrom:
                fieldRef:
                  fieldPath: metadata.namespace
```

Then entry.sh can `kubectl ... -n${NAMESPACE} ...`

Reply:
@JohnStrunk 

> In thinking more about this, for the above to work, we'll need to tell `kubectl` the correct namespace via an env variable:
> 
> ```yaml
>   template:
>     spec:
>       containers:
>           env:
>             - name: NAMESPACE
>               valueFrom:
>                 fieldRef:
>                   fieldPath: metadata.namespace
> ```
> 
> Then entry.sh can `kubectl ... -n${NAMESPACE} ...`

I think we can remove it for now. The [example.jsonnet](https://github.com/gluster/gluster-mixins/blob/master/extras/example.jsonnet) is a dummy file which people can change according to the need(Like if user doesn't want grafana and only need alerts). At that time user can just tweek example.jsonnet file and get the required resources

We added namespace to show what all resources can be configured with example.jsonnet. 

I need to add more documentation in mixins project(how example.jsonnet can be used to generate multiple resources).

Reply:
> Looking at the current container, the `manifests` directory has 3 files that create/modify:
> 
> * A namespace: "monitoring"
> * A ConfigMap: "monitoring/grafana-dashboard-k8s-storage-resources-glusterfs-pv"
> * A monitoring.coreos.com/v1 PrometheusRule: "monitoring/prometheus-alert-rules"
> 
> We're really just adding the ConfigMap and PrometheusRule to the current namespace (this Job is running in the monitoring namespace), so the `00namespace-namespace.yaml` (and associated namespace manipulation perms) shouldn't be needed. The `namespace` fields can also be removed from the other object definitions... This will make it possible to change the namespace the objects get deployed in (determined by the namespace where the Job is run).
> 
> It should then be possible to convert to a Role w/ just '*' on ConfigMap and PrometheusRule.
> 
> (the above will need to be verified, of course... I have likely missed something)

Did the clean up will need to test. Also moved gluster mixins step up in gcs-deploy as grafana expects the mounts to be confiugred first. As it scraps the configmaps mounts which is done in gluster-mixins container.

Reply:
Also sent a PR to remove the creation of monitoring namespace. https://github.com/gluster/gluster-mixins/pull/45

Reply:
Do we still need a ClusterRole or is a Role sufficient?

Reply:
@JohnStrunk does that mean that the `CusterRoleBinding` be converted to `RoleBindings`?

Reply:
Yes.

We should be able to make the move if:
- The resources we need to access are namespaced (i think they are)
- We don't need to access things across namespaces (I believe this is also true... we're just adding objects into our own GCS namespace)

ClusterRole => Role, and ClusterRoleBinding => RoleBinding

Note that the Role becomes a namespaced object whereas the ClusterRole is not